renderdoc/replay/replay_controller.cpp

```diff
@@ -875,8 +875,11 @@ bool ReplayController::SaveTexture(const TextureSave &saveData, const char *path
   if(downcast)
   {
     // if the source and destination are more than 1 byte per component, remap to RGBA32
-    if(td.format.compByteWidth > 1 && (sd.destType == FileType::DDS ||
-                                       sd.destType == FileType::HDR || sd.destType == FileType::EXR))
+    // since D24S8 format has zero compByteWidth, we check by it's type directly
+    if((td.format.compByteWidth > 1 || td.format.type == ResourceFormatType::D24S8) &&
```

Comment:
This should also check for the other packed/special formats that might not have a valid compByteWidth, the other depth/stencil formats at least as well as R10G10B10A2, R11G11B10, and R9G9B9E5.

Reply:
Seems like you're handling those formats directly upon `SaveTexture`.
There's a dedicated check for R10G10B10A2 and R11G11B10 when saving to HDR

(replay_controller.cpp, `if(saveFmt.type == ResourceFormatType::R10G10B10A2)` and `else if(saveFmt.type == ResourceFormatType::R11G11B10)`)

I could add R9G9B9E5 support to it as well if you want, but I believe that'd be better for a separate pull.

Reply:
Ah I forgot that those two formats are manually decoded, good point they don't go through the downcast path.

I'd still prefer to check for D32S8, D16S8, and R9G9B9E5 here. I don't think there's any reason to make it a separate pull request, it's fixing the same bug as you've identified it's just more complete as it covers more formats than the one you ran into.

Reply:
Hi just to check, are you going to update this PR to complete the fix? If not I can make the change.

Reply:
Hi!

Sorry, was a bit busy at work. Yeah, I'll finish this up 🙇

Reply:
No problem! I just wanted to make sure I wasn't waiting on you and you were waiting on me :). Better to ask and be sure.